libs/design/src/core/openable/openable.directive.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  Directive,
+  EventEmitter,
+  HostBinding,
+  Input,
+  Output,
+} from '@angular/core';
+
+import { DaffOpenable } from './openable';
+
+/**
+ * A directive that opens or closes a component. It should only be used as an [Angular Host Directive](https://angular.dev/guide/directives/directive-composition-api).
+ * It supports both a state and stateless implementation, but only one version should be used within a component.
+ *
+ * @usage
+ * ```ts
+ * import {
+ *  Component,
+ *  ChangeDetectionStrategy,
+ * } from '@angular/core';
+ *
+ * import { DaffOpenableDirective } from '@daffodil/design';
+ *
+ * @Component({
+ *   selector: 'custom-component',
+ *   template: `
+ *     <button (click)="toggle()">Click me!</button>
+ *     <div class="hidden-block">This is a hidden block that can be shown by clicking the button.</div>
+ *   `,
+ *   styles: [`
+ *   :host {
+ *     .hidden-block {
+ *       display: none;
+ *     }
+ *
+ *     &.daff-open {
+ *       .hidden-block {
+ *         display: block;
+ *       }
+ *     }
+ *   }`],
+ *   changeDetection: ChangeDetectionStrategy.OnPush,
+ *   hostDirectives: [{
+ *     directive: DaffOpenableDirective,
+ *   }],
+ * })
+ * export class CustomComponent {
+ *   constructor(private openDirective: DaffOpenableDirective) {}
+ *
+ *   toggle() {
+ *     this.openDirective.toggle();
+ *   }
+ * }
+ * ```
+ */
+@Directive({
+  selector: '[daffOpenable]',
+  standalone: true,
+})
+
+export class DaffOpenableDirective implements DaffOpenable {
+  /** Whether or not a component implementing the directive is open */
+  @Input() @HostBinding('class.daff-open') open = false;
```

Comment:
I think there should be an additional guard in the setting of an input to ensure that the internal state is only updated from the bound input in stateless mode.
```suggestion
  private _open = false;
  
  @Input()
  set open(v: boolean) {
    if (this.stateless) {
      this._open = v
    } else {
      // log a warning?
    }
  }
  @HostBinding('class.daff-open')
  get open(): boolean {
    return this._open
  }
```

Reply:
and then in the later methods, only mutate `_open` not `open`

Reply:
made updates to this